### ISSUE TITLE
Add automatic support for UDT

### ISSUE THREAD

**Issue by mmatczuk** [scylladb/gocqlx#120]
Working with UDTs is kind off a pain in gocql. We make it easier with mapper and so on but in general it should be possible to mark struct as UDT and let gocqlx handle that efficiently.

From user perspective enabling UDT should look like

```go
type A struct {
	gocqlx.UDT

	B int
	C string `db:"c2"`
	D MyInt
}
```

Quick sketch for this:

```go

package gocqlx

import (
	"reflect"

	"github.com/gocql/gocql"
)

type UDT struct {
	Ifce map[string]interface{}
}

func (u *UDT) AutoIfce(v interface{}) {
	// Ignore if already set
	if u.Ifce != nil {
		return
	}

	m := DefaultMapper.FieldMap(reflect.ValueOf(v))

	u.Ifce = make(map[string]interface{}, len(m))
	for name, r := range m {
		u.Ifce[name] = r.Interface()
	}
}

func (u UDT) MarshalUDT(name string, info gocql.TypeInfo) ([]byte, error) {
	return gocql.Marshal(info, u.Ifce[name])
}

func (u UDT) UnmarshalUDT(name string, info gocql.TypeInfo, data []byte) error {
	return gocql.Unmarshal(info, data, u.Ifce[name])
}
```

Then it can be hooked to Iterx scanAny and scanAll functions to call `AutoIfce` if `scannable` to init the Ifce map. Note that for scanAll this should be only called once (to avoid repeated reflection calls) and we should use reflect Set (or unsafe pointer directly) to just copy value in the for loop before appending.

**Comment by mmatczuk** [comment]
@martin-sucha please take a look.

**Comment by mmatczuk** [comment]
Fixed in #145
